### ISSUE TITLE
A few questions and a big thanks

### ISSUE THREAD

**Issue by ApeGeneration** [RevenantX/LiteNetLib#213]
I used to work with UNet LLAPI until I came across this last night. I wanted to thank you for making something simple yet effective. After some tinkering I figured out how it works and I can say that I'm very pleased. I don't need no fancy RPC or bloated features no one will ever use, so this is great. There are a few things on my mind that I wanted to ask / comment.

The documentation really needs work. The way it is documented is fine, it's a bit short but that's okay if you understand networking. The examples are clear and the library itself is clean. The main problem is that some of the examples show (im guessing) old ways of doing it. For example: I got stuck on invalid packet because I was using the NetPeer instead of NetPacketProcessor to send custom packets, which obviously did not work! This might be a little nit picking because after some research and looking in the source, I could figure it out using a mix of a few examples.

Next is a question: Is it okay to have multiple listeners? I have loads of different packets that I have to use so I figured it would be better to split it into multiple listeners (for code organization). So would it be okay to have this in my OnNetworkReceive in the server INetEventListener implementing class:
```csharp
listener1.processor.ReadAllPackets(reader, peer);
listener2.processor.ReadAllPackets(reader, peer);
```
Would that cost too much in comparison to having 1 processor delegating all the different packets to the different handlers? Or is this fine for me to use? It would be like 6 listeners at max each with their own processor and the custom packets would be unique to a listener (so there is never more than 1 processor listening for a specific packet using: ``processor.SubscribeReusable<AccountPackage, NetPeer>(OnAccountPackageReceived);``). If there is a better way to solve this or if its needed to make them all go through 1 processor instead of this then I would love to know.

Thanks in advance.

**Library version**: [release version (0.7.7.1)]

**Framework**: [Unity]

**OS**: [Windows]

**Comment by RevenantX** [comment]
> I got stuck on invalid packet because I was using the NetPeer instead of NetPacketProcessor to send custom packets, which obviously did not work! This might be a little nit picking because after some research and looking in the source, I could figure it out using a mix of a few examples.

Yeah... some documentation need updates) Just need enough time for this)

> Next is a question: Is it okay to have multiple listeners?

Multiple listeners is okay) I think it can add little performance impact, because it will check more dictionaries...
But for now `NetPacketProcessor` will throw exceptions if there is unknown packet found (your situation with multiple `NetPacketProcessors`). And first `NetPacketProcessor.ReadAllPackets(...)` will consume `NetDataReader` (it works like stream) so next `NetPacketProcessor` will not work.
So i can change code to support your situation. But better for performance will be 1 NetPacketProcessor. You can bind `SubscribeReusable` to separate classes) I think this will be better.

**Comment by ApeGeneration** [comment]
Such a fast response! Yea I know how painful it can be to write and keep documentation updated haha.

Ah yes, I tested it only with 1 custom packet and 1 listener, I figured I'd ask before I expand and I am glad I did! It is really no problem for me to put it all through 1 processor and delegate it to classes from that, no need to support it since architecture is almost the same anyway (for writing) and performance might become an issue.

Thanks so much for the answer and the nice library, will donate some upcoming month.
